File: hw/Vaca/u2/HW18ZooInPythonJavaScript/Zoo/ec.edu.espe.zoo.model/Monkey.js
const Primate = require('./Primate');

class Monkey extends Primate {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage, tailLength) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
        this.tailLength = tailLength;
    }

    getTailLength() {
        return this.tailLength;
    }

    setTailLength(tailLength) {
        this.tailLength = tailLength;
    }

    toString() {
        return `Monkey{tailLength=${this.tailLength}${super.toString()}}`;
    }
}

module.exports = Monkey;

File: hw/travez/u2/HW18ZooInPython/ec_edu_espe_zoo_model/Monkey.py
import sys
import os

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from ec_edu_espe_zoo_model.Primate import Primate

# Ismael Travez, The Java Bandits, DCCO-ESPE

class Monkey(Primate):
    def __init__(self, id, description, date_onborn, gender, number_of_bones, cage):
        super().__init__(id, description, date_onborn, gender, number_of_bones, cage)

    def feed(self, food):
        print(f"\nFeeding a Monkey with {food} fruits")

